workflow/rules/post-analysis.smk

```diff
@@ -0,0 +1,190 @@
+rule preseq_lc_extrap:
+    input:
+        "results/sam-view/{sample}.bam"
+    output:
+        "results/preseq/{sample}.lc_extrap"
+    params:
+        "-v -seed 1"
+    log:
+        "logs/preseq/{sample}.log"
+    wrapper:
+        "0.64.0/bio/preseq/lc_extrap"
+
+rule collect_multiple_metrics:
+    input:
+         bam="results/orphan_rm_sorted/{sample}.bam",
+         ref="resources/ref/genome.fasta"
+    output:
+        # Through the output file extensions the different tools for the metrics can be selected
+        # so that it is not necessary to specify them under params with the "PROGRAM" option.
+        # Usable extensions (and which tools they implicitly call) are listed here:
+        #         https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/picard/collectmultiplemetrics.html.
+        multiext("{path}{sample}",
+                 ".alignment_summary_metrics",
+                 ".base_distribution_by_cycle_metrics",
+                 ".base_distribution_by_cycle.pdf",
+                 ".insert_size_metrics",
+                 ".insert_size_histogram.pdf",
+                 ".quality_by_cycle_metrics",
+                 ".quality_by_cycle.pdf",
+                 ".quality_distribution_metrics",
+                 ".quality_distribution.pdf"
+                 )
+    resources:
+        # This parameter (default 3 GB) can be used to limit the total resources a pipeline is allowed to use, see:
+        #     https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#resources
+        mem_gb=3
+    log:
+        "logs/picard/{path}{sample}.log"
+    params:
+        # optional parameters
+        "VALIDATION_STRINGENCY=LENIENT "
+    wrapper:
+        "0.64.0/bio/picard/collectmultiplemetrics"
+
+rule genomecov:
+    input:
+        "results/orphan_rm_sorted/{sample}.bam",
+        "results/orphan_rm_sorted/{sample}.orphan_rm_sorted.flagstat"
+    output:
+        pipe("results/bed_graph/{sample}.bedgraph")
+    log:
+        "logs/bed_graph/{sample}.log"
+    params: #-fs option was not used because there are no single end reads any more
+        "-bg -pc -scale $(grep 'mapped (' results/orphan_rm_sorted/{sample}.orphan_rm_sorted.flagstat | awk '{print 1000000/$1}')"
```

Comment:
Move to config.yaml later:
```suggestion
    params:
        # -fs option was not used because there are no single end reads any more
        # TODO: move to config.yaml (including above comment) and load from there
        "-bg -pc -scale $(grep 'mapped (' results/orphan_rm_sorted/{sample}.orphan_rm_sorted.flagstat | awk '{print 1000000/$1}')"
```